src/dialects/postgres/query-generator.js

```diff
@@ -301,8 +301,9 @@ export class PostgresQueryGenerator extends AbstractQueryGenerator {
     const definition = this.dataTypeMapping(table, key, dbDataType);
     const quotedKey = this.quoteIdentifier(key);
     const quotedTable = this.quoteTable(this.extractTableDetails(table));
+    const checkColumnExists = attribute.checkColumnExists;
```

Comment:
IMO it shouldn't go on `ModelAttributeColumnOptions`. It doesn't make much sense to define it on the model because it won't ever have an impact when using models.

This is more for when you use `queryInterface` directly, such as in migrations.

Instead, prefer passing it as part of an option bag:

```ts
// abstract/query-generator.d.ts

export interface AddColumnQueryOptions {
  ifNotExists?: boolean;
}

addColumnQuery(
  table: TableName, 
  columnName: string, 
  columnDefinition: ModelAttributeColumnOptions | DataType,
  options?: AddColumnQueryOptions,
);
```

(could you add this to abstract/query-generator.d.ts?)

I also recommend using `ifNotExists` instead of `checkColumnExists`. It's closer to what you'd use in SQL so less prone to misunderstandings

---

In `abstract/query-interface.js`, you will need to pass the option bag to `addColumnQuery` & update its type definition:

```ts
// abstract/query-interface.d.ts

interface AddColumnOptions extends AddColumnQueryOptions, QueryRawOptions {}

addColumn(
  table: TableName,
  columnName: string,
  columnDefinition: ModelAttributeColumnOptions | DataType,
  options?: QiOptionsWithReplacements
): Promise<void>;
```

Reply:
@ephys I had a question, you wanted to do this?

```ts
// abstract/query-interface.d.ts

interface AddColumnOptions extends AddColumnQueryOptions, QueryRawOptions {}

addColumn(
  table: TableName,
  columnName: string,
  columnDefinition: ModelAttributeColumnOptions | DataType,
  options?: AddColumnOptions // HERE?
): Promise<void>;
```

Second thing I wanted to ask was related to tests, so every database that does not support `IF NOT EXISTS` should throw, however I don't think implementing this in abstract will work so essentially I'll have to check in `addColumnQuery` for each db and then throw if that options is present and not supported. Is that correct or there is a better way around it?

Reply:
Q1: yep :) just like that

Q2: yes that's right. We already do that for some other methods if you want to do something similar: https://github.com/sequelize/sequelize/blob/main/src/dialects/postgres/query-generator.js#L34